shopify_integration/fulfilments.py

```diff
@@ -134,12 +132,25 @@ def create_delivery_notes(
 
 
 def update_fulfillment_items(
-	dn_items: List["DeliveryNoteItem"],
-	fulfillment_items: List["LineItem"]
+	so: "SalesOrder",
+	fulfillment_items: List["LineItem"],
+	dn: "DeliveryNote"
 ):
-	for dn_item in dn_items:
+	for so_item in so.items:
 		# TODO: figure out a better way to add items without setting valuation rate to zero
-		dn_item.allow_zero_valuation_rate = True
 		for item in fulfillment_items:
-			if get_item_code(item) == dn_item.item_code:
-				dn_item.qty = item.attributes.get("quantity")
+			if frappe.db.exists("Delivery Note Item",
+						{"against_sales_order": so_item.name, "shopify_order_item_id": item.attributes.get("id")}):
+				continue
```

Comment:
I'm not sure how this would work. My suggestion was replacing the `get_item_code(...) === so_item_code` if expression.